code/modules/mob/living/simple_animal/friendly/spider.dm

```diff
@@ -0,0 +1,147 @@
+	gold_core_spawnable = FRIENDLY_SPAWN
+	verb_say = "chitters"
+	verb_ask = "chitters inquisitively"
+	verb_exclaim = "chitters intensely"
+	verb_yell = "chitters intensely"
+	faction = list("spiders")
+	del_on_death = TRUE
+	loot = list(/obj/item/reagent_containers/food/snacks/spiderling)
+
+	var/obj/machinery/atmospherics/components/unary/vent_pump/entry_vent
+	var/obj/machinery/atmospherics/components/unary/vent_pump/exit_vent
+	var/travelling_in_vent = 0
+
+/mob/living/simple_animal/spiderling/Initialize()
+	. = ..()
+	pixel_x = rand(6,-6)
+	pixel_y = rand(6,-6)
+	AddComponent(/datum/component/swarming)
+
+/mob/living/simple_animal/spiderling/proc/start_enter_vent()
+	var/list/vents = list()
+	var/datum/pipeline/entry_vent_parent = entry_vent.parents[1]
+	for(var/obj/machinery/atmospherics/components/unary/vent_pump/temp_vent in entry_vent_parent.other_atmosmch)
+		vents.Add(temp_vent)
+	if(!vents.len)
+		entry_vent = null
+		return
+	exit_vent = pick(vents)
+	if(prob(50))
+		visible_message("<span class='warning'>[src] scrambles into the ventilation ducts!</span>", \
+						"<span class='warning'>You hear something scampering through the ventilation ducts.</span>")
+	addtimer(CALLBACK(src, .proc/enter_vent), rand(20,60))
+
+/mob/living/simple_animal/spiderling/proc/enter_vent()
+	travelling_in_vent = TRUE
+	forceMove(entry_vent)
+	var/travel_time = round(get_dist(loc, exit_vent.loc) / 2)
+	addtimer(CALLBACK(src, .proc/travel_vent), travel_time)
+
+/mob/living/simple_animal/spiderling/proc/travel_vent()
+	if(!exit_vent || exit_vent.welded)
+		forceMove(entry_vent)
+		entry_vent = null
+		travelling_in_vent = FALSE
+		return
+	if(prob(50))
+		exit_vent.audible_message("<span class='warning'>You hear something scampering through the ventilation ducts.</span>")
+	forceMove(exit_vent)
+	addtimer(CALLBACK(src, .proc/exit_vent), rand(20,60))
+
+/mob/living/simple_animal/spiderling/proc/exit_vent()
+	if(!exit_vent || exit_vent.welded)
+		forceMove(entry_vent)
+		entry_vent = null
+		travelling_in_vent = FALSE
+		return
+	forceMove(get_turf(exit_vent))
+	travelling_in_vent = FALSE
+	entry_vent = null
+	exit_vent = null
+	var/area/new_area = get_area(src)
+	if(new_area)
+		new_area.Entered(src)
+
+/mob/living/simple_animal/spiderling/Life()
+	..()
+	if(client)
+		return
+
+	if(entry_vent)
+		if(get_dist(src, entry_vent) <= 1)
+			start_enter_vent()
+	//=================
+
+	else if(prob(10))
+		//ventcrawl!
+		for(var/obj/machinery/atmospherics/components/unary/vent_pump/v in view(7,src))
+			if(!v.welded)
+				entry_vent = v
+				walk_to(src, entry_vent, 1)
+				break
+
+//eventually grows into a giant spider
+/mob/living/simple_animal/spiderling/giant
+	var/grow_as = null
+	var/amount_grown = 0
+	var/player_controlled = FALSE
+	var/grown_poison_type = "toxin"
+	var/grown_poison_per_bite = 5
+	var/directive = "" //Message from the mother
+	gold_core_spawnable = NO_SPAWN
+
+/mob/living/simple_animal/spiderling/giant/Life()
+	..()
+	if(isturf(loc))
```

Comment:
is this check still needed?

Reply:
I'm not 100% sure, but i figured that without it you'd get spiders growing in vents.